kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableReadsSuite.scala

```diff
@@ -185,6 +178,45 @@ class DeltaTableReadsSuite extends AnyFunSuite with TestUtils {
     }
   }
 
+  // TODO for the below, when and what error should we throw?
+  //   - on Table creation?
+  //   - on Snapshot creation?
```

Comment:
This is a good point. May be check for `_delta_log` exists or not as part of the `Table` creation, but `_delta_log` could be empty too.

May be it is ok to throw the `TableNotFoundException` later in snapshot creation, but with a different error message (delta_log is not exists or not valid)?

Reply:
https://github.com/delta-io/delta/issues/2253